### PR TITLE
Add fastSAM

### PR DESCRIPTION
**Pull request by Cubevoid**
Not sure what the right way to resize the images is, or to suppress the warnings

### CODE REVIEW

**Comment by Cubevoid** [comment]
TODO: dataloader needs to accept new file format

File: src/data_collection/data_loader.py [hunk 1]
```diff
@@ -1,56 +1,81 @@
...
+            base = start + self.history_len
+            states.append(frames[start:base])
+            obj_bbxs = object_bounding_boxes[base:base+time_steps]  # [T, O, 4]
+            object_bounding_boxes_list.append(obj_bbxs)
+            masks.append(detected_masks[base])
+            actions.append(episode_actions[base:base+time_steps])
+
+        states_tensor = torch.from_numpy(np.array(states))
+        states_tensor  = states_tensor  / 255
+        states_tensor  = states_tensor .permute(0, 4, 1, 2, 3)
```
Comment (quajak) [inline src/data_collection/data_loader.py:65]:
```suggestion
        states_tensor = states_tensor.permute(0, 4, 1, 2, 3)
```

File: src/data_collection/data_loader.py [hunk 2]
```diff
@@ -1,56 +1,81 @@
...
+            start = np.random.randint(0, len(frames) - self.history_len - time_steps)
+            base = start + self.history_len
+            states.append(frames[start:base])
+            obj_bbxs = object_bounding_boxes[base:base+time_steps]  # [T, O, 4]
+            object_bounding_boxes_list.append(obj_bbxs)
+            masks.append(detected_masks[base])
+            actions.append(episode_actions[base:base+time_steps])
+
+        states_tensor = torch.from_numpy(np.array(states))
+        states_tensor  = states_tensor  / 255
```
Comment (quajak) [inline src/data_collection/data_loader.py:64]:
```suggestion
        states_tensor = states_tensor / 255
```

File: src/data_collection/data_loader.py [hunk 3]
```diff
@@ -1,56 +1,81 @@
...
+        states_tensor  = states_tensor .permute(0, 4, 1, 2, 3)
+        object_bounding_boxes_tensor= torch.from_numpy(np.array(object_bounding_boxes_list))
```
Comment (quajak) [inline src/data_collection/data_loader.py:66]:
```suggestion
        object_bounding_boxes_tensor = torch.from_numpy(np.array(object_bounding_boxes_list))
```

**Review by quajak — CHANGES_REQUESTED** [review]

**Review by quajak — APPROVED** [review]

**Review by BENEDIDIN — APPROVED** [review]

**Pull request merged**